tests/testsuite/build_script.rs

```diff
@@ -4921,55 +5021,54 @@ fn rerun_if_directory() {
         )
         .build();
 
-    let dirty = |dirty_line: &str, compile_build_script: bool| {
-        let mut dirty_line = dirty_line.to_string();
-
-        if !dirty_line.is_empty() {
-            dirty_line.push('\n');
-        }
-
-        let compile_build_script_line = if compile_build_script {
-            "[RUNNING] `rustc --crate-name build_script_build [..]\n"
-        } else {
-            ""
-        };
-
-        p.cargo("check -v")
-            .with_stderr(format!(
-                "\
-{dirty_line}\
-[COMPILING] foo [..]
-{compile_build_script_line}\
-[RUNNING] `[..]build-script-build[..]`
-[RUNNING] `rustc --crate-name foo [..]
-[FINISHED] [..]",
-            ))
-            .run();
+    let dirty = |expected| {
+        p.cargo("check -v").with_stderr_data(expected).run();
     };
 
     let fresh = || {
-        p.cargo("check").with_stderr("[FINISHED] [..]").run();
+        p.cargo("check")
+            .with_stderr_data(
+                "\
+[FINISHED] [..]
+",
+            )
+            .run();
```

Comment:
I think this is less important compared to the `dirty`, which I'm currently just leaving as-is.  Let me know if we want to rewrite it with expected argument.